pkg/osquery/runtime/osquery_instance_history/history.go

```diff
@@ -0,0 +1,65 @@
+package osquery_instance_history
+
+import (
+	"errors"
+	"time"
+)
+
+const max_instances = 10
+
+type History struct {
+	instances []*Instance
+}
+
+// GetHistory returns the last 10 instances of osquery started / restarted by launcher
+// each start / restart cycle is an entry
+func GetHistory() []Instance {
+	results := []Instance{}
+	for _, v := range currentHistory.instances {
```

Comment:
I think there are two ways to reduce allocations -- one would be to use set `results` to `len(currentHistory.instances)` is. But I think better is to just bake `max_instances` into the slice lengths.